contributor-docs/adrs/adr-021-automated-prop-docs.md

```diff
@@ -0,0 +1,277 @@
+  - status
+  - a11yReviewed
+- Parse component source code and generate a `components.json` file with data we can use for generating docs ([example components.json](https://unpkg.com/browse/@primer/react@36.24.0/generated/components.json))
+- Test output `components.json` against a schema AND use snapshot testing to ensure we don't introduce regressions
+
+#### Suggested metadata documented with JSDoc tags:
+
+- `@alias`: May be used when the name we use to refer to the component is different from the name React calculates using the component function's name or the `displayName` function property (for example: `Item` being named `ActionList.Item`).
+- `@deprecated`: Marks a component as deprecated.
+- `@primerid`: The unique ID of the component. Not used for subcomponents (for example: `ActionList.Item`).
+- `@primerstatus`: The [lifecycle status](https://primer.style/guides/component-lifecycle) of the component.
+- `@primera11yreviewed`: Whether it has passed a11y design and a11y engineering reviews. If this tag is not set, it defaults to `false`.
+- `@primerparentid`: The ID of the parent component that this subcomponent belongs to. Required for subcomponents (for example: `ActionList.Item`).
+- `@primerstories`: An optional list of Storybook story IDs to override the story IDs we dynamically calculate in the build script by using the component name, status, and file path.
+
+#### Gotchas
+
+If a component's file name doesn't match the Storybook files, our script won't be able to find the Storybook files and parse the data for the `stories` property. For example: ActionList has `List.tsx` and `ActionList.stories.tsx`.
+
+We may be able to fix this by making our story-parsing script a little smarter so it just picks up any `.stories.tsx` files in the directory.
+
+**Workaround:** For the `stories` field to be properly populated, we need the following:
+- A story filed named with the component name. Example: `ActionMenu.stories.tsx`
+- A story named `Default` exported from `{ComponentName}.stories.tsx`
+- (optional) A story filed named `{ComponentName}.features.stories.tsx`
+- (optional) A story filed named `{ComponentName}.examples.stories.tsx`
+
+---
+
+If you want a component to be documented, it must be exported from the `.tsx` file. So far this only causes problems for compount components that we export like this: `export const Root = Object.assign(RootComponent, {Item: ItemComponent})`.
+
+This is a problem with `react-typescript-docgen`. TypeDoc will associate subcomponents as "properties" of the exported parent function component.
+
+**Workarounds:**
+- Instead of `export const Root = Object.assign(RootComponent, {Item: ItemComponent})` in the component's `.tsx` file, export each component from the `.tsx` file and then import into the component's `index.ts` file, then export with `Object.assign` from `index.ts`.
+- Export each subcomponent even if you're not actually importing it anywhere else
+
+#### Other challenges
```

Comment:
For other challenges listed here, what are you thinking would be the best way to format types that may be pulled out in an non-ideal format?

Reply:
Could you give me an example?